src/gobupload/storage/update_table.py

```diff
@@ -0,0 +1,612 @@
+    or_join = ' OR\n    '
+
+    json_join_alias = 'json_arr_elm'
+
+    def __init__(self, src_catalog_name, src_collection_name, src_field_name):
+        self.src_catalog_name = src_catalog_name
+        self.src_collection_name = src_collection_name
+        self.src_field_name = src_field_name
+
+        self.src_collection = self.model.get_collection(src_catalog_name, src_collection_name)
+        self.src_field = self.src_collection['all_fields'].get(src_field_name)
+        self.src_table_name = self.model.get_table_name(src_catalog_name, src_collection_name)
+
+        # Get the destination catalog and collection names
+        self.dst_catalog_name, self.dst_collection_name = self.src_field['ref'].split(':')
+        self.dst_table_name = self.model.get_table_name(self.dst_catalog_name, self.dst_collection_name)
+
+        # Check if source or destination has states (volgnummer, begin_geldigheid, eind_geldigheid)
+        self.src_has_states = self.model.has_states(self.src_catalog_name, self.src_collection_name)
+        self.dst_has_states = self.model.has_states(self.dst_catalog_name, self.dst_collection_name)
+
+        self.relation_specs = self.sources.get_field_relations(src_catalog_name, src_collection_name, src_field_name)
+        if not self.relation_specs:
+            raise RelateException("Missing relation specification for " +
+                                  f"{src_catalog_name} {src_collection_name} {src_field_name} " +
+                                  "(sources.get_field_relations)")
+
+        self.is_many = self.src_field['type'] == "GOB.ManyReference"
+        self.relation_table = "rel_" + get_relation_name(self.model, src_catalog_name, src_collection_name,
+                                                         src_field_name)
+
+    def _select_expressions(self):
+        """Returns the select expressions for the outer query
+
+        :return:
+        """
+        select_expressions = [
+            f'src.{FIELD.ID} AS src__id',
+            f'src.{FIELD.EXPIRATION_DATE} AS src__expiration_date',
+            f"{self._source_value_ref()} AS src_bronwaarde",
+            f"src.{FIELD.SOURCE} AS src__source",
+            f"src.{FIELD.APPLICATION} AS src__application",
+            f"src.{FIELD.SOURCE_ID} AS src__source_id",
+            f"src.{FIELD.VERSION} AS src__version",
+            f"rel.{FIELD.GOBID} AS rel__gobid",
+            f'rel.src_id AS rel_src_id',
+            f'rel.src_volgnummer AS rel_src_volgnummer',
+            f'rel.dst_id AS rel_dst_id',
+            f'rel.dst_volgnummer AS rel_dst_volgnummer',
+            f'rel.{FIELD.EXPIRATION_DATE} AS rel__expiration_date',
+            f'rel.{FIELD.ID} AS rel__id',
+            f'src.{FIELD.SOURCE} AS src_source',
+            f'dst.{FIELD.ID} AS dst__id',
+            f'dst.{FIELD.EXPIRATION_DATE} AS dst__expiration_date',
+            f'LEAST(src.{FIELD.EXPIRATION_DATE}, dst.{FIELD.EXPIRATION_DATE}) AS expected_expiration_date'
+        ]
```

Comment:
why having src__source and src_source in the same array for the same field?
I propose to keep all src_, rel_ and dst_ together. Now it changes from src to rel to dst to src again